### PR TITLE
WIP: Style updates to Settings

### PR DESCRIPTION
**Pull request by bsclifton**
## Current status
This is something I've worked on as time allows. Feel free to grab it and help! Feedback appreciated

This PR is meant to fix:
https://github.com/brave/brave-browser/issues/958
https://github.com/brave/brave-browser/issues/964

And partially addresses:
https://github.com/brave/brave-browser/issues/1222
https://github.com/brave/brave-browser/issues/1186
https://github.com/brave/brave-browser/issues/960
https://github.com/brave/brave-browser/issues/955
https://github.com/brave/brave-browser/issues/26
(and possibly others)

Here's a screenshot showing how it looks at the moment:
screen shot 2018-10-12 at 12 46 41 pm

## TODOs
- [x] Remove the hamburger menu which doesn't work
- [ ] Use correct font (Poppins)
- [ ] Put the icons in the title for each section
- [x] Add the new shields section to sidebar
- [x] Wire up the shields link, so it jumps down there (routable URL like `chrome://settings/shields`)
- [x] Fix display problem with SVGs (See picture. Was it a bad export maybe?)
- [ ] Create `Get Started` section and condense appropriate settings there
- [ ] Use same orange header color on Bookmarks page
- [ ] Use same orange header color on History page
- [ ] Use same orange header color on Downloads page

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by rossmoody** [comment]
Did some light CSS and icon render fixes. Latest screen: 
image

**Comment by bsclifton** [comment]
Just finished rebasing this and making sure it works great against master 😄 👍

File: patches/chrome-browser-resources-settings-settings_icons_css.html.patch [hunk 1]
```diff
@@ -0,0 +1,27 @@
+diff --git a/chrome/browser/resources/settings/settings_icons_css.html b/chrome/browser/resources/settings/settings_icons_css.html
+index 7a4e72d47e8e79a89aacd2f84fa47575d122e0a8..5a4c8832a7b91499f19b4d08468639f1dc453383 100644
+--- a/chrome/browser/resources/settings/settings_icons_css.html
++++ b/chrome/browser/resources/settings/settings_icons_css.html
+@@ -4,14 +4,14 @@
+ <dom-module id="settings-icons">
+   <template>
+     <style>
+-<if expr="chromeos">
```
Comment (bridiver) [inline patches/chrome-browser-resources-settings-settings_icons_css.html.patch]: I'm confused about this patch. This only applies to chromeos

File: patches/chrome-browser-resources-settings-settings_menu-settings_menu.html.patch [hunk 2]
```diff
@@ -1,28 +1,232 @@
...
+ 
++<if expr="not _google_chrome">
++<link rel="import" href="../brave_settings_menu/icons.html">
++</if>
++
+ <dom-module id="settings-menu">
+   <template>
+     <style include="settings-shared">
+       :host {
+-        display: block;
```
Comment (bridiver) [inline patches/chrome-browser-resources-settings-settings_menu-settings_menu.html.patch]: we need to look at other ways to do this, this patch is going to be a nightmare to maintain

**Comment by bridiver** [comment]
we definitely need to look at other ways to handle these patches, they are going to be effectively unmaintainable across chromium upgrades

File: patches/ui-webui-resources-cr_elements-cr_toolbar-cr_toolbar.html.patch [hunk 3]
```diff
@@ -0,0 +1,22 @@
+diff --git a/ui/webui/resources/cr_elements/cr_toolbar/cr_toolbar.html b/ui/webui/resources/cr_elements/cr_toolbar/cr_toolbar.html
+index 8fb18010132c64a0ed6ff19357e43642ad9c2311..fcf7ae4023558853f62a8231c17db8620741726b 100644
+--- a/ui/webui/resources/cr_elements/cr_toolbar/cr_toolbar.html
++++ b/ui/webui/resources/cr_elements/cr_toolbar/cr_toolbar.html
+@@ -133,7 +133,7 @@
+     <div id="leftContent">
+       <div id="leftSpacer">
+         <!-- Note: showing #menuPromo relies on this dom-if being [restamp]. -->
+-        <template is="dom-if" if="[[showMenu]]" restamp>
++        <!-- <template is="dom-if" if="[[showMenu]]" restamp>
```
Comment (bridiver) [inline patches/ui-webui-resources-cr_elements-cr_toolbar-cr_toolbar.html.patch]: if this is based on `showMenu` then why don't we just turn off `showMenu`?

File: patches/chrome-browser-ui-webui-settings-about_handler.cc.patch [hunk 4]
```diff
@@ -0,0 +1,21 @@
+diff --git a/chrome/browser/ui/webui/settings/about_handler.cc b/chrome/browser/ui/webui/settings/about_handler.cc
+index c372368a1e0abe7d9ee2af13208f328ea4416c1f..240de00ea312f01ea82028e8af3ed83bf82a22b3 100644
+--- a/chrome/browser/ui/webui/settings/about_handler.cc
++++ b/chrome/browser/ui/webui/settings/about_handler.cc
+@@ -282,6 +282,16 @@ AboutHandler::~AboutHandler() {
+ 
+ AboutHandler* AboutHandler::Create(content::WebUIDataSource* html_source,
+                                    Profile* profile) {
++  std::string brave_product_version = version_info::GetVersionNumber();
```
Comment (bridiver) [inline patches/chrome-browser-ui-webui-settings-about_handler.cc.patch]: I believe this logic already exists somewhere and if it doesn't it needs to be put in a helper. It definitely doesn't belong here in this patch

File: patches/chrome-browser-resources-settings-settings_icons_css.html.patch (hunk 1 above)
Comment (petemill) [reply patches/chrome-browser-resources-settings-settings_icons_css.html.patch]: looks like just an indentation thing? I can fix it

File: browser/resources/settings/brave_settings_menu/icons.html [hunk 5]
```diff
@@ -0,0 +1,36 @@
+<link rel="import" href="../html/polymer.html">
```
Comment (petemill) [inline browser/resources/settings/brave_settings_menu/icons.html]: Leaving a note here to remember to change this path to point at the path in chromium's resources

File: patches/ui-webui-resources-cr_elements-cr_toolbar-cr_toolbar.html.patch (hunk 3 above)
Comment (petemill) [reply patches/ui-webui-resources-cr_elements-cr_toolbar-cr_toolbar.html.patch]: showMenu is a function on the prototype for MenuButton. So I think we'll have to extend and set to undefined.

**Comment by Jacalz** [comment]
Really looking forward to seeing this merged. This looks really promising :+1:

**Comment by bsclifton** [comment]
Closing in favor of the branch @petemill has been working on 😄

**Pull request closed without merging**